### PR TITLE
Build s390x again

### PR DESCRIPTION
**Pull request by StefanScherer**
Signed-off-by: Stefan Scherer <stefan.scherer@docker.com>

### CODE REVIEW

**Comment by thaJeztah** [comment]
@StefanScherer can you rebase?

**Comment by StefanScherer** [comment]
Rebased. Wow still a lot of failures. 🤔

**Comment by thaJeztah** [comment]
ah, yes it's now failing on the man pages; that will be fixed by https://github.com/docker/containerd-packaging/pull/165

**Comment by thaJeztah** [comment]
https://github.com/docker/containerd-packaging/pull/165 was merged; I'll try kicking CI again (I think it does a merge before building), but otherwise another rebase may be needed 🤞

**Comment by thaJeztah** [comment]
Man pages are still failing, so looks like a rebase is needed


Fedora 30 is failing on aarch64; perhaps those repos are no longer up, or is it expecting (e.g.) `arm64` or `armv8` for the repository URLs?

```
#7 [fedora-base 2/2] RUN dnf install -y rpm-build git dnf-plugins-core
#7 29.13 Fedora 30 - aarch64 - Updates                    21 kB/s | 424 kB     00:20
#7 29.13 Errors during downloading metadata for repository 'updates':
#7 29.13   - Status code: 404 for https://pubmirror2.math.uh.edu/fedora-buffet/fedora/linux/updates/30/Everything/aarch64/repodata/2f0ee96dd70baf7454cb79f51f1509037f12a43ece1edb9a01bc645b9de5230a-filelists.xml.zck (IP: 129.7.128.190)
#7 29.13   - Status code: 404 for https://mirror.its.sfu.ca/mirror/fedora/linux/updates/30/Everything/aarch64/repodata/87090d179476b9edd218586793d6e11600a95bafdcae48d662416bc7d0e20d0f-updateinfo.xml.zck (IP: 142.58.101.156)
#7 29.13   - Status code: 404 for http://mirror.math.princeton.edu/pub/fedora/linux/updates/30/Everything/aarch64/repodata/efab835db9fcf7f891218d562fbe827c3c1c93a323ad4d21c4e4ea16caf5a7f6-prestodelta.xml.zck (IP: 128.112.18.21)
...
```


Hm, nope, that's not it; looks like the ARCH is ok, but the URL's for those `zck` files is incorrect. On Fedora 31, I'm also seeing these;

```
#7 [fedora-base 2/2] RUN dnf install -y rpm-build git dnf-plugins-core
#7 30.52 Fedora 31 - aarch64 - Updates                    23 kB/s | 425 kB     00:18    
#7 30.53 Errors during downloading metadata for repository 'updates':
#7 30.53   - Status code: 404 for https://mirror.arizona.edu/fedora/linux/updates/31/Everything/aarch64/repodata/ba7024a7f52c0b07c33d7a7fca17723f521ecac341bb6a2e78bd2760e6cf4444-updateinfo.xml.zck (IP: 128.196.204.204)
#7 30.53   - Status code: 404 for https://dl.fedoraproject.org/pub/fedora/linux/updates/31/Everything/aarch64/repodata/84c89111c05f462bdf2e56d4b051bef93aad732092e3d9d16aa896a762118ed2-prestodelta.xml.zck (IP: 209.132.181.23)
#7 30.53   - Status code: 404 for http://pubmirror1.math.uh.edu/fedora-buffet/fedora/linux/updates/31/Everything/aarch64/repodata/84c89111c05f462bdf2e56d4b051bef93aad732092e3d9d16aa896a762118ed2-prestodelta.xml.zck (IP: 129.7.128.189)
...
```

And looking at that location, that file has a different name; perhaps it's a case of mirrors being out of date?

http://repo.atlantic.net/fedora/linux/updates/31/Everything/aarch64/repodata/

File: Jenkinsfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ def images = [
     [image: "balenalib/rpi-raspbian:stretch", arches: ["armhf"]],
     [image: "balenalib/rpi-raspbian:buster",  arches: ["armhf"]],
     [image: "ubuntu:xenial",                  arches: ["amd64", "aarch64", "armhf"]], // Ubuntu 16.04 LTS (End of support: April, 2021. EOL: April, 2024)
-    [image: "ubuntu:bionic",                  arches: ["amd64", "aarch64", "armhf"]], // Ubuntu 18.04 LTS (End of support: April, 2023. EOL: April, 2028)
+    [image: "ubuntu:bionic",                  arches: ["amd64", "aarch64", "armhf", "s390x"]], // Ubuntu 18.04 LTS (End of support: April, 2023. EOL: April, 2028)
     [image: "ubuntu:disco",                   arches: ["amd64", "aarch64", "armhf"]], // Ubuntu 19.03  (EOL: January, 2020)
     [image: "ubuntu:eoan",                    arches: ["amd64", "aarch64", "armhf"]], // Ubuntu 19.10  (EOL: July, 2020)
     [image: "ubuntu:focal",                   arches: ["amd64", "aarch64"]],          // Ubuntu 20.04 LTS (End of support: April, 2025. EOL: April, 2030)
```
Comment (thaJeztah) [inline Jenkinsfile:25]: Should we enable it for all ubuntu versions?

Comment (StefanScherer) [reply Jenkinsfile:25]: Good catch. That was for the first test builds. I'll add the others as well.

Comment (StefanScherer) [reply Jenkinsfile]: Added it to debian and ubuntu versions.

File: common/common.mk [hunk 2]
```diff
@@ -23,6 +23,11 @@ GOVERSION?=$(shell curl -fsSL "https://raw.githubusercontent.com/containerd/cont
 endif
 
 GOLANG_IMAGE=golang:$(GOVERSION)
+ifeq ($(OS),Windows_NT)
+       GOLANG_IMAGE=docker.io/library/golang:$(GOVERSION)
+else
+       GOLANG_IMAGE=docker.io/library/golang:$(GOVERSION)-buster
```
Comment (StefanScherer) [inline common/common.mk:29]: This is a workaround for golang:1.12.x images  where the manifest list is missing s390x. With golang:1.13.x or newer this if block can ve removed again.

Comment (justincormack) [reply common/common.mk:29]: can we make tis conditional on Go 1.12.x as well then? We are in process of switching as 1.12 is out of support.

**Pull request merged**